qiskit_experiments/curve_analysis/models.py

```diff
@@ -0,0 +1,354 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=abstract-method, no-member
+
+"""Fit model wrapper of LMFIT to support multi-objective optimization."""
+
+import time
+import warnings
+from typing import Callable, List, Dict, Tuple, Sequence, Iterator
+
+import numpy as np
+import uncertainties
+from asteval import Interpreter, get_ast_names
+from lmfit.minimizer import Minimizer
+from lmfit.model import Model
+from lmfit.parameter import Parameters
+
+from qiskit_experiments.curve_analysis.curve_data import SeriesDef, SolverResult
+from qiskit_experiments.exceptions import AnalysisError
+
+
+class CurveModel(Model):
+    """A model that describes the curve fitting function.
+
+    The model manages generation of the fitting function, computation of
+    residual value against given parameters as a cost function of optimization,
+    and preparation of parameter objects passed to the solver instance.
+    The :class:`CurveModel` is implemented on top of the `LMFIT`_ ``Model`` class
+    with some extension for the multi objective optimization
+    described in :ref:`curve_analysis_overview`.
```

Comment:
```suggestion
    The model manages the generation of the fitting function, computes the
    residual value against the given parameters as a cost function of optimization,
    and prepares the parameter objects passed to the solver instance.
    The :class:`CurveModel` is implemented on top of the `LMFIT`_ ``Model`` class
    with extensions for the multi objective optimization
    described in :ref:`curve_analysis_overview`.
```
This `computes the residual value against the given parameters as a cost function of optimization` is unclear do you have a better wording for it?

Reply:
Custom model class and solver are removed for simplification.